File: DataStructures/table/TestTable/unittest1.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "table.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace TestTable
{		
	TEST_CLASS(UnitTest1)
	{
	public:
		
		TEST_METHOD(AddElementAndFind)
		{
			Table<int, double> t;
			t.AddElement(1,2.5);
			Assert::AreEqual(t.Find(1),2.5);
		}

		TEST_METHOD(AddElement2)
		{
			Table<int, double> t;
			t.AddElement(1,1);
			t.AddElement(2,2);
			t.AddElement(3,3);
			Assert::IsTrue(t.Find(1) == 1 && t.Find(2) == 2 && t.Find(3) == 3);
		}

		TEST_METHOD(AddElement3)
		{
			Table<int, double> t;
			t.AddElement(1,1);
			t.AddElement(1,2);
			Assert::AreEqual(t.Find(1),1.0);
		}

		TEST_METHOD(Replace)
		{
			Table<int, double> t;
			t.AddElement(1,2.5);
			t.Replace(1,-2.5);
			Assert::AreEqual(t.Find(1),-2.5);
		}

		TEST_METHOD(Erase)
		{
			Table<int, double> t;
			t.AddElement(1,2.5);
			t.Replace(1,-2.5);
			Assert::AreEqual(t.Find(1),-2.5);
		}

		TEST_METHOD(Find)
		{
			Table<int, double> t;
			Assert::ExpectException<logic_error>
			(
				[&] 
				{
					t.AddElement(1, 1.0);
					t.Erase(1);
					t.Find(1);
				}
			);
		}

	};
}

File: DataStructures/myString/UnitTestsMyString/UnitTest1.cs
namespace UnitTestsMyString
{
    using System;
    using Microsoft.VisualStudio.TestTools.UnitTesting;
    using myString;

    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethodConcatenation()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inB.txt");
            Assert.AreEqual((a+b).ToString(), "qwerty1234567890qwerty");
        }

        [TestMethod]
        public void TestMethodEquel1()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inC.txt");
            Assert.IsTrue(a == b);
        }

        [TestMethod]
        public void TestMethodEquel2()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inB.txt");
            Assert.IsFalse(a == b);
        }

        [TestMethod]
        public void TestMethodNotEquel1()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inB.txt");
            Assert.IsTrue(a != b);
        }

        [TestMethod]
        public void TestMethodNotEquel2()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inC.txt");
            Assert.IsFalse(a != b);
        }

        [TestMethod]
        public void TestMethodFind1()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inD.txt");
            Assert.AreEqual(a.Find(b), 4);
        }

        [TestMethod]
        public void TestMethodFind2()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inB.txt");
            Assert.AreEqual(a.Find(b), -1);
        }

        [TestMethod]
        public void TestMethodReplace1()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            MyString c = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inD.txt");
            c.GetLine("inE.txt");
            Assert.AreEqual(a.Replace(b, c).ToString(), "qwer12ty3");
        }

        [TestMethod]
        public void TestMethodReplace2()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            MyString c = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inD.txt");
            c.GetLine("inF.txt");
            Assert.AreEqual(a.Replace(b, c).ToString(), "qwer03");
        }

        [TestMethod]
        public void TestMethodReplace3()
        {
            MyString a = new MyString();
            MyString b = new MyString();
            MyString c = new MyString();
            a.GetLine("inA.txt");
            b.GetLine("inD.txt");
            c.GetLine("inG.txt");
            Assert.AreEqual(a.Replace(b, c).ToString(), "qwer-2-103");
        }

        [TestMethod]
        public void TestMethodIndex1()
        {
            MyString a = new MyString();
            a.GetLine("inA.txt");
            Assert.AreEqual(a[0], 'q');
        }

        [TestMethod]
        public void TestMethodIndex2()
        {
            MyString a = new MyString();
            a.GetLine("inA.txt");
            a[0] = '0';
            Assert.AreEqual(a.ToString(), "0werty123");
        }

        [TestMethod]
        public void TestCopyConstructor()
        {
            MyString a = new MyString();
            a.GetLine("inA.txt");
            MyString b = new MyString(a);
            Assert.IsTrue(a == b);
        }
    }
}
